Data/Scripts/001_Technical/002_RubyUtilities.rb

```diff
@@ -162,6 +159,147 @@ def self.move(source, destination)
   end
 end
 
+#===============================================================================
+#  Extensions for the `Color` class
+#===============================================================================
+class Color
+  # alias for old constructor
+  alias init_original initialize unless self.private_method_defined?(:init_original)
+  #-----------------------------------------------------------------------------
+  # new constructor accepts RGB values as well as a hex number or string value
+  #-----------------------------------------------------------------------------
+  def initialize(*args)
+  	pbPrintException("Wrong number of arguments! At least 1 is needed!") if args.length < 1
+  	if args.length == 1
+      if args.first.is_a?(Fixnum)
+        hex = args.first.to_s(16)
+      elsif args.first.is_a?(String)
+        try_rgb_format = args.first.split(',')
+        return init_original(*try_rgb_format) if try_rgb_format.length.between?(3, 4)
+        hex = args.first.delete('#')
+      end
+      pbPrintException("Wrong type of argument given!") if !hex
+      r = hex[0...2].to_i(16)
+      g = hex[2...4].to_i(16)
+      b = hex[4...6].to_i(16)
+  	elsif args.length == 3
+      r, g, b = *args
+  	end
+  	return init_original(r, g, b) if r && g && b
+  	return init_original(*args)
+  end
+  #-----------------------------------------------------------------------------
+  # returns an RGB color value as a hex value
+  #-----------------------------------------------------------------------------
+  def to_hex
+  	r = sprintf("%02X", self.red)
+  	g = sprintf("%02X", self.green)
+  	b = sprintf("%02X", self.blue)
+  	return ("#" + r + g + b).upcase
+  end
```

Comment:
Shouldn't this (and `def initialize` above) support alpha as well if it's not the default 255? Or should there be a similar but separate method that also includes the alpha value in the returned hex string? Or maybe this method should have a boolean parameter that determines whether it includes the alpha in the returned string?

Reply:
There's no alpha in hex notation. Hence why I didn't want to put in an alpha as an additional argument since it would make no sense in that notation form. `color.alpha = value` can still be used after the initial color object is created.